# 01. Importing Libraries

In [37]:
# Import libraries
import pandas as pd
import numpy as np
import os

# 02. Importing Data

In [38]:
# Import the data set into Jupyter as a new dataframe
path = r'/Users/ChuahLH/Desktop/Documents/Instacart Basket Analysis'

In [39]:
# Import the orders_products_combined dataframe
df_orders_products = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'orders_products_combined.pkl'))

In [40]:
# Investigate the orders_product_combined
df_orders_products.head()

Unnamed: 0  order_id  user_id  order_number  order_dow  order_hour_of_day  \
0           0   2539329        1             1          2                  8   
1           0   2539329        1             1          2                  8   
2           0   2539329        1             1          2                  8   
3           0   2539329        1             1          2                  8   
4           0   2539329        1             1          2                  8   

   days_since_last_order  product_id  add_to_cart_order  reordered _merge  
0                    NaN       196.0                1.0        0.0   both  
1                    NaN     14084.0                2.0        0.0   both  
2                    NaN     12427.0                3.0        0.0   both  
3                    NaN     26088.0                4.0        0.0   both  
4                    NaN     26405.0                5.0        0.0   both

In [41]:
# Investigate the shape of the orders_product_combined
df_orders_products.shape

(32640698, 11)

In [42]:
# Import the product_data sets
df_prods_clean = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'products_clean_data.csv'), index_col = False)

In [43]:
# Investigate the products_clean_data
df_prods_clean.head()

Unnamed: 0  product_id                                       product_name  \
0           0           1                         Chocolate Sandwich Cookies   
1           1           2                                   All-Seasons Salt   
2           2           3               Robust Golden Unsweetened Oolong Tea   
3           3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...   
4           4           5                          Green Chile Anytime Sauce   

   aisle_id  department_id  prices  
0        61             19     5.8  
1       104             13     9.3  
2        94              7     4.5  
3        38              1    10.5  
4         5             13     4.3

In [44]:
# Investigate the shape of products_clean_data
df_prods_clean.shape

(49677, 6)

There are 32640698 data under the orders_product_combined; 49677 data under the product_id. Merge the data using product_id. 

# 03. Merge Data

In [45]:
# There is a _merge column orders_products_combined. Drop the _merge column.
df_orders_products = df_orders_products.drop(columns = ['_merge'])

In [46]:
# Merged the orders_products_combined and product_clean
df_orders_products_merged = df_orders_products.merge(df_prods_clean, on = 'product_id', indicator = True)

In [47]:
df_orders_products_merged.head()

Unnamed: 0_x  order_id  user_id  order_number  order_dow  \
0             0   2539329        1             1          2   
1             1   2398795        1             2          3   
2             2    473747        1             3          3   
3             3   2254736        1             4          4   
4             4    431534        1             5          4   

   order_hour_of_day  days_since_last_order  product_id  add_to_cart_order  \
0                  8                    NaN       196.0                1.0   
1                  7                   15.0       196.0                1.0   
2                 12                   21.0       196.0                1.0   
3                  7                   29.0       196.0                1.0   
4                 15                   28.0       196.0                1.0   

   reordered  Unnamed: 0_y product_name  aisle_id  department_id  prices  \
0        0.0           195         Soda        77              7     9.0   
1        1.0           195         Soda        77              7     9.0   
2        1.0           195         Soda        77              7     9.0   
3        1.0           195         Soda        77              7     9.0   
4        1.0           195         Soda        77              7     9.0   

  _merge  
0   both  
1   both  
2   both  
3   both  
4   both

In [48]:
# Check the merged data
df_orders_products_merged['_merge'].value_counts()

both          32406041
left_only            0
right_only           0
Name: _merge, dtype: int64

In [49]:
# Export the dataframe in pkl format
df_orders_products_merged.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'orders_products_merged.pkl'))